apps-devstg/us-east-1/alb-logs-s3-bucket/bucket.tf

```diff
@@ -0,0 +1,56 @@
+#
+# Create an S3 bucket for storing ALB access logs
+#
+module "s3_bucket_alb_logs" {
+  source = "github.com/binbashar/terraform-aws-s3-bucket.git?ref=v3.15.2"
+
+  bucket = "${local.bucket_name}-logs"
+  acl    = "log-delivery-write"
+  # Allow deletion of non-empty bucket
+  force_destroy = true
+
+  control_object_ownership = true
+  object_ownership         = "ObjectWriter"
+
+  attach_elb_log_delivery_policy = true # Required for ALB logs
+  attach_lb_log_delivery_policy  = true # Required for ALB/NLB logs
+
+  lifecycle_rule = [
+    {
+      id      = "billing-objects-logs"
+      enabled = true
+      prefix  = "logs/"
+
+      tags = {
+        rule      = "log"
+        autoclean = "true"
+      }
+
+      transition = [
+        {
+          days          = 30
+          storage_class = "ONEZONE_IA"
+          }, {
+          days          = 180
+          storage_class = "GLACIER"
+        }
+      ]
+
+      expiration = {
+        days = 365
+      }
```

Comment:
Just curious: do we really need this long retention for these logs? Where does that requirement come from? Business? Compliance?

Reply:
I'll move this definition to `locals.tf` to make it more easy to configure as required.